packages/nvidia_nat_core/src/nat/middleware/timeout/timeout_middleware.py

```
@@ -0,0 +1,120 @@
+
+        Raises:
+            TimeoutError: If the downstream call exceeds the configured timeout.
+        """
+        timeout: float = self._timeout_config.timeout
+        try:
+            return await asyncio.wait_for(
+                super().function_middleware_invoke(*args, call_next=call_next, context=context, **kwargs),
+                timeout=timeout,
+            )
+        except TimeoutError:
+            logger.error("Function '%s' exceeded timeout of %ss", context.name, timeout)
```

Ruff (0.15.2)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)